### ISSUE TITLE
moveEnvelopes returns undefined instead of a promise

### ISSUE THREAD

**Issue by mdhornet90** [docusign/docusign-node-client#124]
Trying to use the `moveEnvelopes` api function, I discovered that the promisified version always returned `undefined`. This seems to be because of code in `ApiClient.js` regarding the return strategy.

Old callback style:
```
request.end(function(error, response) {
  if (!error) {
    try {
      data = _this.deserialize(response, returnType);
    } catch (err) {
      error = err;
    }
  }
  callback(error, data, response);
});
```

Promisified results:
```
return new Promise(function(resolve, reject){
  request.end(function(error, data) {
    if (error) {
      reject(error);
    } else {
      try {
        data = _this.deserialize(data, returnType);
        resolve(data);
      } catch(error) {
        reject(error);
      }
    }
  })
});
```
In the case of `moveEnvelopes`, `data` never seems to be successfully deserialized (not sure if this is intentional), and so is always `undefined`.

**Comment by acooper4960** [comment]
Hi @mdhornet90 this is because the methods return a promise themselves, as opposed to a request object. Here is a sample call as outlined in the new readme.

`   envelopesApi.createEnvelope(accountId, {'envelopeDefinition': envDef})
      .then(function(envelopeSummary) {
        console.log('EnvelopeSummary: ' + JSON.stringify(envelopeSummary));
        return next(null);
      })
      .catch(function (error){
        if (error) {
          return next(error);
        }
      });`

**Comment by acooper4960** [comment]
Hi @mdhornet90 this is because the methods return a promise themselves, as opposed to a request object. Here is a sample call as outlined in the new readme. Promises follow a different pattern than error-first callbacks. Instead of retrieving the error in your callback, you can catch it with Promise.catch

envelopesApi.createEnvelope(accountId, {'envelopeDefinition': envDef}) .then(function(envelopeSummary) { console.log('EnvelopeSummary: ' + JSON.stringify(envelopeSummary)); return next(null); }) .catch(function (error){ if (error) { return next(error); } });

**Comment by mdhornet90** [comment]
Maybe this isn't necessarily a problem with Promises directly, but my use of the Promise-based API for `moveEnvelopes` results in a response type of `undefined`. This is also true of the callback-style API, for some reason deserialization of the response always yields it.

**Comment by acooper4960** [comment]
Hmm, have you tried catching any thrown exceptions? The Promise should be returned before deserialization occurs

**Comment by acooper4960** [comment]
Would it be possible to send the code of your api call? Also, have you tried catching any thrown errors?

**Comment by LarryKlugerDS** [comment]
Hi @mdhornet90,
Thank you for the bug report. I investigated and found that the underlying problem is that the SDK is expecting a response body from the API call. The reality is that this API method returns a 0 length response. 

The fix (within DocuSign) is to correct the Swagger file that is used to mechanically generate the SDK and then regenerate and test the SDK's method. I have filed bug report DCM-3085. 

## Workaround
You could call the API directly (not use the SDK).

**Comment by LarryKlugerDS** [comment]
I will rename this issue to more closely reflect the underlying problem.

**Comment by acooper4960** [comment]
Hi @mdhornet90 this issue has been resolved in the upcoming v4.3.0. This was actually related to incorrect documentation, stating that this method should return a value when it should not. Feel free to download the rc from npm via: npm install docusign-esign@next. Please let us know if this resolves your issue/
